### ISSUE TITLE
Attribute modifiers

### ISSUE THREAD

**Issue by kshvmdn** [kshvmdn/fsql#17]
A lot of suggestions have revolved around modifying attributes in some way (#3, #15).

Attribute modifiers will apply _some change_ to the attributes in the `SELECT` and `WHERE` clauses.

I'm most fond of the following syntax right now. Of the ideas suggested, it's the easiest to parse (particularly for chained modifiers), while still being intuitive. I'm open to ideas if anyone has a suggestion.

```sql
SELECT name::upper::fullpath, size::mb FROM . WHERE time::iso > 2017-05-17T06:07:28Z
```

As I mentioned in https://github.com/kshvmdn/fsql/issues/15#issuecomment-301967875, one of the downsides of a syntax like this is that we're moving away from SQL (i'm not entirely sure that this is even a problem).

**Comment by jorbs** [comment]
Why not use them as functions? E.g.

`SELECT fullpath(upper(name)), format(size, 'mb') FROM . WHERE format(time, 'iso') > 2017-05-17T06:07:28Z`

**Comment by kshvmdn** [comment]
@jorbs I had originally considering using function syntax as well, but I decided against it for a few reasons --

1) Parsing chained functions is _slightly_ more work (and messier).
2) Invoking fsql without quotes requires you to escape parentheses, so to run the above example, you'd use:

    ```sh
    $ fsql SELECT fullpath\(upper\(name\)\), format\(size, 'mb'\) FROM . WHERE format\(time, 'iso'\) \> 2017-05-17T06:07:28Z
    ```
    
    Of course, you still have the option of wrapping the entire query in quotes and not having to escape anything.

That being said, it seems like most people do prefer function syntax, so I'll probably end up going with that anyways! :smile:

**Comment by khageshpatel** [comment]
Hi,

I should have asked you before working on this. I was just playing around the code of this and implemented something of this sort. Please see if its of any use or not: [#23](https://github.com/kshvmdn/fsql/pull/23)

**Comment by kshvmdn** [comment]
Hey @khageshpatel, no worries! I'll get to reviewing and testing that right away!

**Comment by kshvmdn** [comment]
Completed in #27 (09ea293713036930690a399e4b6951901de7c19f).

I ended up opting with the function notation for this to be consistent with SQL syntax. I also introduced an "interactive" mode in hopes of mitigating the need for escaping parentheses (which was my main concern with this notation, as mentioned above).

See [README.md#attribute-modifiers](https://github.com/kshvmdn/fsql#attribute-modifiers) for the full documentation.
